plugins/plugin_utils/keys_filter.py

```diff
@@ -92,17 +92,49 @@ def _keys_filter_target_str(target, matching_parameter):
 
 
 def _keys_filter_target_dict(target, matching_parameter):
-    """test:
-    * target is a list of dictionaries
-    * ...
+    """
+       Test:
+       * target is a list of dictionaries with attributes 'after' and 'before'.
+       * Attributes 'before' must be valid regex if matching_parameter=regex.
+       * Otherwise, the attributes 'before' must be strings.
+       Convert target and return:
+       * iterator that aggregates attributes 'before' and 'after', or
+       * iterator that aggregates compiled regex of attributes 'before' and 'after' if matching_parameter=regex.
     """
 
-    # TODO: Complete and use this in filter replace_keys
+    if not isinstance(target, list):
+        msg = "The target must be a list. target is %s."
+        raise AnsibleFilterError(msg % (target, type(target)))
 
-    if isinstance(target, list):
-        for elem in target:
-            if not isinstance(elem, Mapping):
-                msg = "The target items must be dictionary. %s is %s"
-                raise AnsibleFilterError(msg % (elem, type(elem)))
+    if len(target) == 0:
+        msg = "The target can't be empty."
+        raise AnsibleFilterError(msg)
 
-    return
+    for elem in target:
+        if not isinstance(elem, Mapping):
+            msg = "The target items must be dictionaries. %s is %s"
+            raise AnsibleFilterError(msg % (elem, type(elem)))
+        if not all(k in elem for k in ('before', 'after')):
+            msg = "All dictionaries in target must include: after, before."
+            raise AnsibleFilterError(msg)
+
+    before = [d['before'] for d in target]
+    after = [d['after'] for d in target]
+
+    if matching_parameter == 'regex':
+        try:
+            tr = map(re.compile, before)
+            tz = list(zip(tr, after))
+        except re.error:
+            msg = ("The attributes before must be valid regex if matching_parameter=regex."
+                   " Not all items are valid regex in: %s")
+            raise AnsibleFilterError(msg % before)
+    else:
+        if not all(isinstance(item, string_types) for item in before):
```

Comment:
I would move this check up (checking for string before compiling a regex is a good idea as well), and add a similar check for `after`.

Reply:
Done. I put these tests into the above iteration of the target. Thank you for the review!